app/controllers/admin/ingredients_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Admin::IngredientsController < ApplicationController
+  http_basic_authenticate_with name: "admin", password: "1234"
```

Comment:
you should take those info from ENV[''], and set in production using `heroku config:set`